src/python_testing/TC_AVSM_2_2.py

```diff
@@ -0,0 +1,196 @@
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --endpoint 1
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class TC_AVSM_2_2(MatterBaseTest):
+    def desc_TC_AVSM_2_2(self) -> str:
+        return "[TC-AVSM-2.2] Validate Snapshot Stream Allocation functionality with Server as DUT"
+
+    def pics_TC_AVSM_2_2(self):
+        return ["AVSM.S"]
+
+    def steps_TC_AVSM_2_2(self) -> list[TestStep]:
+        return [
+            TestStep("precondition", "Commissioning, already done", is_commissioning=True),
+            TestStep(
+                1, "TH reads FeatureMap attribute from CameraAVStreamManagement Cluster on TH_SERVER", "Verify SNP is supported"
+            ),
+            TestStep(
+                2,
+                "TH reads AllocatedSnapshotStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated snapshot streams in the list is 0.",
+            ),
+            TestStep(
+                3,
+                "TH reads SnapshotCapabilities attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aSnapshotCapabilities.",
+            ),
+            TestStep(
+                4,
+                "TH sends the SnapshotStreamAllocate command with valid values of ImageCodec, MaxFrameRate, MinResolution=MaxResolution=Resolution from aSnapshotCapabilities and Quality set to 90.",
+                "DUT responds with SnapshotStreamAllocateResponse command with a valid SnapshotStreamID.",
+            ),
+            TestStep(
+                5,
+                "TH reads AllocatedSnapshotStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated snapshot streams in the list is 1.",
+            ),
+            TestStep(
+                6,
+                "TH sends the SnapshotStreamAllocate command with values from step 3 except with MaxFrameRate set to 0(outside of valid range).",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+            TestStep(
+                7,
+                "TH sends the SnapshotStreamAllocate command with values from step 3 except with Quality set to 101(outside of valid range).",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+            TestStep(
+                8,
+                "TH sends the SnapshotStreamAllocate command with values from step 3 except with Quality set to 101(outside of valid range).",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+        ]
+
```

Comment:
There's new decorators which enure this TC is only run when the cluster is present on the endpoint, otherwise what can happen is this is executed, and fails, on an EP where the cluster isn't actually present.
Suggest:
from chip.testing.matter_testing import has_cluster, run_if_endpoint_matches
...
@run_if_endpoint_matches(has_cluster(Clusters.CameraAvStreamManagement))

Reply:
Thank you for your review. We fixed it with you comments "run_if_endpoint_matches"